API.md

```diff
@@ -0,0 +1,121 @@
+API Proposal
+============
+## Use Cases
+- **Application Administration** (e.g. GPII): updating multiple settings of multiple (other) applications
+- **Application Settings** (e.g. my app): read/update/monitor applications's own settings
+
+
+## To throw or not to throw
```

Comment:
I think throwing is a reasonable behaviour in modern JavaScript. The days when runtimes mishandled exceptions are long-lost in the mists of the IE8 era. The alternative is to return rejected promises, which ties you to a particular promise handling library - this seems less desirable to me.